dart/lib/src/metrics/local_metrics_aggregator.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:core';
+import 'package:meta/meta.dart';
+import '../protocol/metric_summary.dart';
+import 'metric.dart';
+
+@internal
+class LocalMetricsAggregator {
+  // format: <export key, <metric key, gauge>>
+  final Map<String, Map<String, GaugeMetric>> _buckets = {};
+
+  void add(final Metric metric, final num value) {
+    final bucket =
+        _buckets.putIfAbsent(metric.getSpanAggregationKey(), () => {});
+
+    bucket.update(metric.getCompositeKey(), (m) => m..add(value),
+        ifAbsent: () => Metric.fromType(
+            type: MetricType.gauge,
```

Comment:
Because this is an aggregation per span. It aggregates the values of each metric emitted while the span runs [under gauges](https://develop.sentry.dev/sdk/metrics/#span-aggregation), so that in the dashboard (Sentry.io) you can see min, max, count and avg of each metric.